tapscript/tx.go

```diff
@@ -111,7 +111,28 @@ func virtualTxIn(newAsset *asset.Asset, prevAssets commitment.InputSet) (
 
 // virtualTxOut computes the single output of a Taproot Asset virtual
 // transaction based on whether an asset has a split commitment or not.
-func virtualTxOut(txAsset *asset.Asset) (*wire.TxOut, error) {
+func virtualTxOut(txAsset *asset.Asset, sighashFlag txscript.SigHashType) (
+	*wire.TxOut, error) {
+
+	// If the SigHashNone flag is used, we don't need to commit to any
+	// output, so we just create an empty tree.
+	if sighashFlag == txscript.SigHashNone {
```

Comment:
Alternatively, what stops us from just relying on Bitcoin here? We can make the same output as normal, but then we sign it using sighash none. So everything is still committed to as normal, but Bitcoin level part is what blanks out output when signing.

Reply:
Yeap! Makes sense